gCoffee/app/src/main/java/com/mborowiec/gcoffee/Order.java

```diff
@@ -106,10 +107,23 @@ private void makeOrder(String coffee) {
                 Toast.LENGTH_SHORT).show();
     }
 
+    /**
+     * Checks whether the menu contains the ordered item. If it does, asks to confirm order.
+     * @param data incoming deeplink
+     */
     private void handleDeepLinks(Uri data) {
-        String coffee;
-        coffee = data.getQueryParameter("inMenuName");
-        showConfirmationDialog(coffee);
+            String coffee;
+            coffee = data.getQueryParameter("inMenuName");
+            String[] coffeeList = this.getResources().getStringArray(R.array.coffee_list);
+            boolean contains = Arrays.asList(coffeeList).contains(coffee);
```

Comment:
I'm trying to understand potential pitfalls here ;-) What are potential reasons why the coffee is not on the menu?

Reply:
What about a more explicit test for invalid input, e.g., the following would work in the future with a refactored "toLower" string.

coffee = data.getQueryParameter("inMenuName");
if(coffee == null) {
    // Unexpected input
    // TODO fail gracefully
}

String[] coffeeList = this.getResources().getStringArray(R.array.coffee_list);
boolean contains = Arrays.asList(coffeeList).contains(coffee.toLower());

Reply:
What about extracting the "domain logic" and adding a test for the different failure cases?